libs/libc/stdlib/lib_getprogname.c

```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+ * libs/libc/stdlib/lib_getprogname.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/tls.h>
+#include <stdlib.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: getprogname
+ ****************************************************************************/
+
+FAR const char *getprogname(void)
+{
+  FAR struct task_info_s *info;
+
+  info = task_get_info();
+  return info->argv[0];
```

Comment:
```suggestion
#ifndef CONFIG_BUILD_KERNEL
  FAR struct task_info_s *info;

  info = task_get_info();
  DEBUGASSERT(info != NULL);
  return (info->argv && info->argv[0]) ? info->argv[0] : "?";
#else
  return "?";
#endif
```

Reply:
We don't check CONFIG_BUILD_KERNEL here, since argv field always exist in all three mode(flat, protected and kernel).